imagehash.go

```diff
@@ -104,3 +112,86 @@ func (h *ImageHash) ToString() string {
 	}
 	return fmt.Sprintf(strFmt, kindStr, h.hash)
 }
+
+// Implement ExtImageHash from here
+func newExtImageHash(hash []uint64, kind Kind) *ExtImageHash {
+	return &ExtImageHash{hash: hash, kind: kind}
+}
+
+// Distance method returns a distance between two big hashes
+func (h *ExtImageHash) Distance(other *ExtImageHash) (int, error) {
+	if h.kind != other.kind {
+		return -1, errors.New("Extended Image hashes's kind should be identical")
+	}
+
+	lHash := h.hash
+	rHash := other.hash
+	if len(lHash) != len(rHash) {
+		return -1, errors.New("Extended Image hashes's size should be identical")
+	}
+
+	var distance int 
+	for idx, lh := range lHash {
+		rh := rHash[idx]
+		hamming := lh ^ rh
+		distance += popcnt(hamming)
+	}	
+	return distance, nil
+}
+
+// ToString returns a hex representation of big hash
+func (h *ExtImageHash) ToString() string {
+	var hexBytes []byte
+	for _, hash := range h.hash {
+		hashBytes := make([]byte, 8)
+		binary.BigEndian.PutUint64(hashBytes, hash)
+		hexBytes = append(hexBytes, hashBytes...)
+	}
+	return hex.EncodeToString(hexBytes)
+}
+
+// GetKind method returns a kind of big hash
+func (h *ExtImageHash) GetKind() string {
+	kindStr := ""
+	switch h.kind {
+	case AHash:
+		kindStr = "a"
+	case PHash:
+		kindStr = "p"
+	case DHash:
+		kindStr = "d"
+	case WHash:
+		kindStr = "w"
+	}
+	return kindStr
+}
+
+// ExtImageHashFromString returns a big hash from a hex representation
+func ExtImageHashFromString(s, kindStr string) (*ExtImageHash, error) {
```

Comment:
I'd like to align to XXXHashFromString API with https://github.com/corona10/goimagehash/pull/4